source/dub/init.d

```diff
@@ -158,7 +158,20 @@ private void initDeimosPackage(Path root_path, ref PackageRecipe p, scope void d
 private void writeGitignore(Path root_path)
 {
 	write((root_path ~ ".gitignore").toNativeString(),
-		".dub\ndocs.json\n__dummy.html\n*.o\n*.obj\n__test__*__\n");
+q"{.dub
+docs.json
+__dummy.html
+docs/
+*.so
+*.dylib
+*.dll
+*.a
+*.lib
```

Comment:
I have some repositories that have library files checked in, so I think this is too broad as a default. It would definitely make sense to ignore "(project name).{a,lib,dll,dylib,exe}".